app/client/ui/DocumentSettings.ts

```diff
@@ -298,7 +408,15 @@ export class DocSettingsPage extends Disposable {
   }
 }
 
-
+function persistType(type: string|null, docId: string|undefined){
+  docId = docId?.split("~")[0];
+  return fetch(`/o/docs/api/docs/${docId}`,
+    { method:'PATCH',
+      headers: {"Content-Type": "application/json"},
+      credentials: 'include',
+      body:JSON.stringify({type})
+    }).catch((err)=>{ console.log(err); });
```

Comment:
Is the `console.log` intentional here?

Reply:
No, I think it's better to call the logger here.

Reply:
I would say that leaving the promise uncaught should make an error notification appear at the bottom-right corner.

https://github.com/gristlabs/grist-core/blob/356f0b423ea786efe2f52f1bb4c9f206fa515a49/app/client/models/errors.ts#L196-L199

Reply:
With this diff:

```diff
diff --git a/app/client/ui/DocumentSettings.ts b/app/client/ui/DocumentSettings.ts
index 302114ca..03d24e54 100644
--- a/app/client/ui/DocumentSettings.ts
+++ b/app/client/ui/DocumentSettings.ts
@@ -335,11 +335,8 @@ export class DocSettingsPage extends Disposable {
         } else {
           docType = "Tutorial";
         }
-        persistType(docType, docId)
-        .then(()=>window.location.reload())
-        .catch(err=>console.log(err));
-
-        ctl.close();
+        await persistType(docType, docId);
+        window.location.reload();
       };
 
       const documentTypeOptions = () => [
@@ -415,7 +412,7 @@ function persistType(type: string|null, docId: string|undefined){
       headers: {"Content-Type": "application/json"},
       credentials: 'include',
       body:JSON.stringify({type})
-    }).catch((err)=>{ console.log(err); });
+    });
 }
```

Which:
 - leaves the modal open while the request has not succeeded;
 - does not close the modal but instead let the refresh make the modal disappear;
 - doesn't handle directly here the error, leaving the handler above catch and report the error;

This is how it renders:

![2024-09-05_18-58](https://github.com/user-attachments/assets/f6d16fb1-b4a9-4914-b2b0-a360b7cc1533)